src/solve/test.rs

```diff
@@ -567,6 +760,46 @@ fn normalize_implied_bound() {
     }
 }
 
+#[test]
+fn normalize_implied_bound_gat() {
```

Comment:
Pre-existing: could this be renamed to just "gat_implied_bounds" or something like this?